CMakeLists.txt

```diff
@@ -161,4 +161,39 @@ if( T8CODE_BUILD_FORTRAN_INTERFACE )
     endif()
 endif()
 
+# Mark all P4EST and SC options as advanced such that they are not shown in the default cmake GUIs.
+mark_as_advanced( FORCE
+    P4EST_ENABLE_BUILD_2D
+    P4EST_ENABLE_BUILD_3D
+    P4EST_ENABLE_BUILD_P6EST
+    P4EST_ENABLE_FILE_CHECKS
+    P4EST_ENABLE_MPI
+    P4EST_ENABLE_MPICOMMSHARED
+    P4EST_ENABLE_MPIIO
+    P4EST_ENABLE_MPITHREAD
+    P4EST_ENABLE_MPIWINSHARED
+    P4EST_ENABLE_VTK_COMPRESSION
+    P4EST_HAVE_FSYNC
+    P4EST_HAVE_GETOPT_H
+    P4EST_HAVE_INTTYPES_H
+    P4EST_HAVE_JSON
+    P4EST_HAVE_MEMORY_H
+    P4EST_HAVE_POSIX_MEMALIGN
+    P4EST_HAVE_STDINT_H
+    P4EST_HAVE_STDLIB_H
+    P4EST_HAVE_STRING_H
+    P4EST_HAVE_SYS_STAT_H
+    P4EST_HAVE_SYS_TYPES_H
+    P4EST_HAVE_UNISTD_H
+    P4EST_HAVE_ZLIB
+    P4EST_NEED_M
+    P4EST_USE_SYSTEM_SC
+    SC_BUILD_SHARED_LIBS
+    SC_ENABLE_OPENMP
+    SC_HAVE_JSON
+    SC_TEST_WITH_VALGRIND
+    SC_USE_INTERNAL_JSON
+    SC_USE_INTERNAL_ZLIB
```

Comment:
We could run into problems as soon as p4est or sc changes its build variables. What do you think about automatically grabbing all those options? Alternative solution: https://github.com/DLR-AMR/t8code/pull/1570